### ISSUE TITLE
Exclude list

### ISSUE THREAD

**Issue by jfastnacht** [simov/markdown-viewer#6]
Would be great to have an exclude list, like it's implemented on plugins like AdBlock "Exclude this site/domain".

**Comment by simov** [comment]
That's a problem only when the web site in question serves the files with their raw name, meaning they end in `.md`.

Currently these domains are excluded:

```js
"exclude_matches": [
  "*://github.com/*",
  "*://bitbucket.org/*",
  "*://gitlab.com/*"
]
```

So if you navigate to https://github.com/simov/markdown-viewer/blob/master/README.md you can see that the extension is not enabled there.

If you want to see the *raw* version of that file https://raw.githubusercontent.com/simov/markdown-viewer/master/README.md you will notice that the extension is enabled there because this domain is not excluded. You can still view the raw markdown if you choose to do so from the extension's menu.

Can you tell me which web site you have problems with? And what exactly those problems are.

**Comment by jfastnacht** [comment]
Yes, I know, that's exactly the problem. Happened to me on https://getcomposer.org/doc/00-intro.md . Awesome extension, btw. :+1:

**Comment by simov** [comment]
The excluded domains above serve their files with an `.md` extension but they actually serve rendered html instead. So excluding them was critical because it led to some weird visual bugs when the extension tryied to render them on top of the html.

The link you posted on the other hand falls in that second category where it is up to the user to choose whether he want to see the raw markdown or rendered html, and that option is stored.

I understand that it can be annoying if you regularly switch between different tabs that require either rendered html or raw markdown, but I don't see it as a huge problem at this point, unless there is strong user demand for that feature. I'll definitely add it to my todo list for future reference.

**Comment by jfastnacht** [comment]
The https://getcomposer.org/doc/00-intro.md page is serving HTML with a `.md` extension too, afaik. Opening it with my Mozilla Firefox without any extensions works and serves HTML, but opening it with Chrome and your extension only serves:

![image](https://cloud.githubusercontent.com/assets/6183607/18471004/be93a97a-79af-11e6-9da0-580b5171f073.png)

In your definition that would mean this page is critical too and my proposal was meant for critical pages rather than predefinitions for Raw or HTML view.

**Comment by simov** [comment]
Yes, you are correct, I didn't checked it in another browser. I will add the `getcomposer.org` to the exclude list, thanks for reporting it :+1: 

Do you want me to publish new release soon?

**Comment by simov** [comment]
Actually I just released version 2.1, but it may take up to an hour before it become available. I guess you'll receive the update automatically after that.

Closing this issue now. Thanks for contributing!

BTW I'm starting to think that an options page won't be that bad after all.
